app/src/main/java/com/ad340/whatdo/PickerUtils.java

```diff
@@ -0,0 +1,93 @@
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Locale;
+
+public class PickerUtils {
+
+    public static DatePickerDialog.OnDateSetListener onDateSetListener ( // RecyclerViewAdapter
+            Calendar c, Todo todo, ToDoItemRecyclerViewAdapter.ToDoItemViewHolder holder,
+            OnTodoInteractionListener listener) {
+        return (view, year, monthOfYear, dayOfMonth) -> {
+            c.set(Calendar.YEAR, year);
+            c.set(Calendar.MONTH, monthOfYear);
+            c.set(Calendar.DAY_OF_MONTH, dayOfMonth);
+
+            StringBuilder dateString = new StringBuilder(DateFormat.getDateInstance(DateFormat.FULL).format(c.getTime()));
+
+            holder.toDoDate.setText(dateString);
+            listener.onUpdateTodo(todo, String.valueOf(dateString), Constants.DATE);
+
+        };
+    }
+
+    public static DatePickerDialog.OnDateSetListener onDateSetListener (  // MainActivity
+            Calendar c, StringBuilder dateString) {
+        return (view, year, monthOfYear, dayOfMonth) -> {
+            c.set(Calendar.YEAR, year);
+            c.set(Calendar.MONTH, monthOfYear);
+            c.set(Calendar.DAY_OF_MONTH, dayOfMonth);
+
+            dateString.append(DateFormat.getDateInstance(DateFormat.FULL).format(c.getTime()));
+        };
+    }
+
+    public static void showDatePicker(Context context, Calendar c, ImageButton imageButton,
+                                      DatePickerDialog.OnDateSetListener date) {
+        imageButton.setOnClickListener(view ->
+                new DatePickerDialog(context, date,
+                        c.get(Calendar.YEAR),
+                        c.get(Calendar.MONTH),
+                        c.get(Calendar.DAY_OF_MONTH)).show()
+        );
+    }
+
+
+    public static TimePickerDialog.OnTimeSetListener onTimeSetListener ( // RecyclerViewAdapter
+            Calendar c, Todo todo, ToDoItemRecyclerViewAdapter.ToDoItemViewHolder holder,
+            OnTodoInteractionListener listener) {
+        return (view, hourOfDay, minute) -> {
+            c.set(Calendar.HOUR_OF_DAY, hourOfDay);
+            c.set(Calendar.MINUTE, minute);
+
+            SimpleDateFormat sdf = new SimpleDateFormat("h:mm a", Locale.US);
+            String timeString = sdf.format(c.getTime());
+
+            holder.toDoTime.setText(timeString);
+            listener.onUpdateTodo(todo, timeString, Constants.TIME);
+
+        };
+    }
+
+    public static TimePickerDialog.OnTimeSetListener onTimeSetListener ( // MainActivity
+            Calendar c, StringBuilder timeString) {
+        return (view, hourOfDay, minute) -> {
+            c.set(Calendar.HOUR_OF_DAY, hourOfDay);
+            c.set(Calendar.MINUTE, minute);
+
+            SimpleDateFormat sdf = new SimpleDateFormat("h:mm a", Locale.US);
+            timeString.append(sdf.format(c.getTime()));
+        };
+    }
+
+
+    public static void showTimePicker(Context context, Calendar c, ImageButton imageButton,
```

Comment:
Any thoughts on changing the showPicker method names to setTimePickerShowOnClick or something like that? Cause it seems like it's just the tiniest bit confusing to have a method called show that actually attaches an onClickListener. Is that just me?

Reply:
Happy to go with whatever is easiest to understand for all involved! I'll change it 👌

Reply:
Updated